client/src/features/visualizer-threejs/ConfigControls.tsx

```diff
@@ -0,0 +1,236 @@
+    const setForcedZoom = useTangleStore((state) => state.setForcedZoom);
+    const [localZoom, setLocalZoom] = useState<number | undefined>(forcedZoom);
+
+    const [state, setState] = useState<IControlsVisualiser>(() => {
+        // Use getFromLocalStorage to retrieve the state
+        return getFromLocalStorage() || defaultControlsVisualiser;
+    });
+    const [isShowReset, setIsShowReset] = useState(() => {
+        return isExistsInLocalStorage();
+    });
+
+    const [errors, setErrors] = useState<{
+        [k: string]: string;
+    }>({});
+
+    const inputs: {
+        key: TKey;
+        label: string;
+        min: number;
+        max: number;
+    }[] = [
+        {
+            key: "MIN_SINUSOID_PERIOD",
+            label: "Min sinusoid period",
+            min: 1,
+            max: 7,
+        },
+        {
+            key: "MAX_SINUSOID_PERIOD",
+            label: "Max sinusoid period",
+            min: 8,
+            max: 15,
+        },
+        {
+            key: "MIN_SINUSOID_AMPLITUDE",
+            label: "Min sinusoid amplitude",
+            min: 100,
+            max: 199,
+        },
+        {
+            key: "MAX_SINUSOID_AMPLITUDE",
+            label: "Max sinusoid amplitude",
+            min: 200,
+            max: 500,
+        },
+        {
+            key: "MIN_TILT_FACTOR_DEGREES",
+            label: "Min tilt factor degrees",
+            min: 1,
+            max: 15,
+        },
+        {
+            key: "MAX_TILT_FACTOR_DEGREES",
+            label: "Max tilt factor degrees",
+            min: 16,
+            max: 100,
+        },
+        {
+            key: "TILT_DURATION_SECONDS",
+            label: "Tilt_duration_seconds",
+            min: 1,
+            max: 10,
+        },
+    ];
+
+    const handleApply = () => {
+        if (Object.keys(errors).some((key) => errors[key])) {
+            // Handle the error case, e.g., display a message
+            console.error("There are errors in the form.");
+            return;
+        }
+
+        setToLocalStorage(state);
+        location.reload();
+    };
+
+    const handleChange = (key: TKey, val: string) => {
+        const input = inputs.find((input) => input.key === key);
+        if (!input) return;
+
+        if (!val) {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: "Value is required" }));
+            setState((prevState) => ({ ...prevState, [key]: "" }));
+            return;
+        }
+
+        const numericValue = Number(val);
+        if (numericValue < input.min || numericValue > input.max) {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: `Value must be between ${input.min} and ${input.max}` }));
+        } else {
+            setErrors((prevErrors) => ({ ...prevErrors, [key]: "" }));
+        }
+
+        setState((prevState) => ({ ...prevState, [key]: numericValue }));
+    };
+
+    if (!features.controlsVisualiserEnabled) {
+        return null;
+    }
+
+    return (
```

Comment:
since this component has it's own scss file, i recommend moving the inline styles there